https://bugs.gentoo.org/933120
https://invent.kde.org/plasma/kwin/-/merge_requests/5892

From b9ec66615f08e2bad3928b8b207be904ab62e745 Mon Sep 17 00:00:00 2001
From: Sam James <sam@gentoo.org>
Date: Sun, 2 Jun 2024 00:25:26 +0100
Subject: [PATCH] watchdog: add includes for geteuid

05a3e2bad9a3baf27293f0b5ed4bef4f38952804 added `geteuid` use but didn't
add the needed headers. This works by chance on libstdc++ systems via
transitive includes, but it doesn't work on libc++ systems, as reported
downstream in Gentoo at https://bugs.gentoo.org/933120.

(cherry picked from commit 31a3961c4cd387aa4aa1a0c7bf197445a346e5f0)
--- a/src/watchdog.cpp
+++ b/src/watchdog.cpp
@@ -8,6 +8,8 @@
 */
 
 #include "watchdoglogging.h"
+#include <unistd.h>
+#include <sys/types.h>
 #include <QCoreApplication>
 #include <QTimer>
 #include <systemd/sd-daemon.h>
-- 
GitLab
